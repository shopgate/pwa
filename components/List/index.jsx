import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import BaseList from '@shopgate/pwa-common/components/List';
import BaseListItem from '@shopgate/pwa-common/components/List/components/Item';
import Item from './components/Item';
import styles from './style';

/**
 * The list component.
 */
class List extends Component {
  static Item = Item;

  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    hasImages: PropTypes.bool,
  };

  static defaultProps = {
    children: null,
    className: '',
    hasImages: false,
  };

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    const { className, children, hasImages } = this.props;

    if (!React.Children.count(children)) {
      return null;
    }

    return (
      <BaseList className={classNames(className, styles.container)}>
        {React.Children.map(children, (child, index) => {
          if (!React.isValidElement(child)) {
            return null;
          }

          // The key for each child.
          const key = `child-${index}`;

          return (
            <BaseListItem
              className={hasImages ? styles.itemWithImage : null}
              isSelected={child.props.isSelected}
              key={key}
            >
              <div className={styles.item}>
                {child}
              </div>
            </BaseListItem>
          );
        })}
      </BaseList>
    );
  }
}

export default List;
