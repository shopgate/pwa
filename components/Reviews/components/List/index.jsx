import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Title from './components/Title';
import Rating from './components/Rating';
import Text from './components/Text';
import Info from './components/Info';
import styles from './style';

/**
 * Review List Component.
 */
class List extends PureComponent {
  static propTypes = {
    reviews: PropTypes.arrayOf(PropTypes.shape()),
  };

  static defaultProps = {
    reviews: null,
  };

  /**
   * @returns {JSX}
   */
  render() {
    const { reviews } = this.props;

    if (!reviews || reviews.length === 0) {
      return null;
    }

    return reviews.map(review => (
      <div key={review.id} className={styles} data-test-id={`reviewTitle: ${review.title}`}>
        <Title title={review.title} />
        <Rating rate={review.rate} />
        <Text review={review.review} />
        <Info review={review} />
      </div>
    ));
  }
}

export default List;
