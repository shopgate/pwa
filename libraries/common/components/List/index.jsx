import React from 'react';
import PropTypes from 'prop-types';
import ListItem from './components/Item';

/**
 * The List component.
 * @param {Object} props The component props.
 * @returns {JSX}
 */
const List = ({ children, className, role }) => (
  <ul className={`${className} common_list`} role={role}>{children}</ul>
);

List.Item = ListItem;

List.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.oneOfType([PropTypes.string, PropTypes.shape()]),
  role: PropTypes.string,
};

List.defaultProps = {
  className: '',
  role: null,
};

export default List;
