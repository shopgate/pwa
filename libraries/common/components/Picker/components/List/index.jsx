import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './style';

/**
 * The default button for the Picker component.
 * @returns {JSX} The button component.
 */
const PickerList = ({
  items,
  onClose,
  onSelect,
  selectedIndex,
}) => (
  <ul className="common__picker__list">
    {items.map((item, currentIndex) => (
      <li
        key={item.value}
        className={classNames({ [styles.active]: currentIndex === selectedIndex })}
      >
        <button
          className={styles.button}
          disabled={item.disabled}
          onClick={() => {
            onSelect(item.value);
            onClose();
          }}
          type="button"
        >
          {item.label}
        </button>
      </li>
    ))}
  </ul>
);

PickerList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  onSelect: PropTypes.func.isRequired,
  onClose: PropTypes.func,
  selectedIndex: PropTypes.number,
};

PickerList.defaultProps = {
  onClose: () => {},
  selectedIndex: null,
};

export default PickerList;
