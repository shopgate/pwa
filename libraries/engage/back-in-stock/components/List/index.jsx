import React, { useEffect, useState } from 'react';
import { css } from 'glamor';
import { LoadingIndicator } from '@shopgate/pwa-ui-shared';
import {
  Accordion, Card,
} from '@shopgate/engage/components';
import { useBackInStockSubscriptionsContext } from '../../hooks';
import Subscription from '../Subscription';
import { i18n } from '../../../core';

const styles = {
  divider: css({
    height: 1,
    width: 'calc(100% + 32px)',
    backgroundColor: 'rgb(234, 234, 234)',
    marginLeft: -16,
    marginRight: -16,
    marginBottom: 16,
  }).toString(),
};

/**
 * The BackInStockReminders component.
 * @returns {JSX}
 */
const List = () => {
  const [isInitial, setIsInitial] = useState(true);

  const {
    subscriptions,
    isFetching,
  } = useBackInStockSubscriptionsContext();

  useEffect(() => {
    if (!isFetching && isInitial) {
      setIsInitial(false);
    }
  }, [isFetching, isInitial]);

  const groupedSubscriptions = subscriptions.reduce((acc, subscription) => {
    const { status } = subscription;
    const groupingStatus = (status === 'inactive' || status === 'triggered') ? 'past' : status;
    acc[groupingStatus].push(subscription);
    return acc;
  }, {
    active: [],
    past: [],
  });

  return (
    <div>
      {Object.entries(groupedSubscriptions).map(([groupKey, filteredSubscriptions]) => (
        <Card className={styles.root} key={groupKey}>
          <Accordion
            className=""
            openWithChevron
            renderLabel={() => groupKey}
            chevronPosition="left"
            startOpened
          >
            <div className={styles.divider} />
            {isInitial ? <LoadingIndicator /> : null}
            {!isInitial && filteredSubscriptions.length === 0 ? (
              <span>{i18n.text('favorites.empty')}</span>
            ) : null}

            {!isInitial && filteredSubscriptions.map((subscription, index) => (
              <div>
                <Subscription subscription={subscription} key={subscription.subscriptionCode} />
                {(index === filteredSubscriptions.length - 1) ?
                  null :
                  <div className={styles.divider} />}
              </div>
            ))}
          </Accordion>
        </Card>
      ))}

    </div>
  );
};

export default List;
