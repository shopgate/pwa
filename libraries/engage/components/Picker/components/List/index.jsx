import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './style';

/**
 * The default button for the Picker component.
 * @param {Object} props The component props.
 * @returns {JSX} The button component.
 */
const PickerList = (props) => {
  const {
    items, onClose, onSelect, selectedIndex, query,
  } = props;

  const filteredItems = useMemo(() => {
    if (!query || !query.length) {
      return items;
    }

    return items.filter(({ label = '' }) => {
      const searchLabel = label.replaceAll(' ', '').toLowerCase();
      const searchValue = query.replaceAll(' ', '').toLowerCase();
      return searchLabel.includes(searchValue);
    });
  }, [items, query]);
  return (
    <ul className="engage__picker_list">
      {filteredItems.map((item, currentIndex) => (
        <li
          key={item.value}
          className={classNames({
            [styles.active]: currentIndex === selectedIndex,
          })}
        >
          <button
            className={styles.button}
            disabled={item.disabled}
            onClick={() => {
              onSelect(item.value);
              onClose();
            }}
            type="button"
          >
            {item.label}
          </button>
        </li>
      ))}
    </ul>
  );
};

PickerList.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  onSelect: PropTypes.func.isRequired,
  onClose: PropTypes.func,
  query: PropTypes.string,
  selectedIndex: PropTypes.number,
};

PickerList.defaultProps = {
  onClose: () => { },
  selectedIndex: null,
  query: '',
};

export default PickerList;
