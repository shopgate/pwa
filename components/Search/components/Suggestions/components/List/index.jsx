import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { SurroundPortals } from '@shopgate/engage/components';
import {
  SEARCH_SUGGESTIONS,
  SEARCH_SUGGESTION_ITEM,
  SEARCH_SUGGESTION_ITEM_CONTENT,
} from '@shopgate/engage/search';
import connect from './connector';
import styles from './style';

/**
 * The SuggestionList component.
 */
class SuggestionList extends Component {
  static propTypes = {
    onClick: PropTypes.func.isRequired,
    fetching: PropTypes.bool,
    suggestions: PropTypes.arrayOf(PropTypes.string),
  };

  static defaultProps = {
    fetching: false,
    suggestions: [],
  };

  /**
   * @param { Object } nextProps Next props.
   * @return {boolean}
   */
  shouldComponentUpdate(nextProps) {
    return nextProps.fetching === false && nextProps.suggestions;
  }

  /**
   * @return {JSX}
   */
  render() {
    const { onClick, suggestions } = this.props;

    if (!suggestions) {
      return null;
    }

    return (
      <SurroundPortals
        portalName={SEARCH_SUGGESTIONS}
        portalProps={{
          onClick,
          suggestions,
        }}
      >
        <div className={styles.list}>
          {suggestions.map(suggestion => (
            <SurroundPortals
              portalName={SEARCH_SUGGESTION_ITEM}
              portalProps={{
                className: styles.item.toString(),
                onClick,
                suggestion,
              }}
              key={suggestion}
            >
              <button
                type="button"
                className={styles.item}
                onClick={onClick}
                value={suggestion}
                data-test-id={`searchSuggestion ${suggestion}`}
              >
                <SurroundPortals
                  portalName={SEARCH_SUGGESTION_ITEM_CONTENT}
                  portalProps={{ suggestion }}
                >
                  {suggestion}
                </SurroundPortals>
              </button>
            </SurroundPortals>
          ))}
        </div>
      </SurroundPortals>
    );
  }
}

export { SuggestionList as UnwrappedSuggestionList };

export default connect(SuggestionList);
