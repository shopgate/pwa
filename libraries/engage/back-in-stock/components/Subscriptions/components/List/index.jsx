import React from 'react';
import { css } from 'glamor';
import LoadingIndicator from '@shopgate/pwa-ui-shared/LoadingIndicator';
import { Accordion, Card } from '@shopgate/engage/components';
import { i18n } from '@shopgate/engage/core';
import { useBackInStockSubscriptionsContext } from '../../../../hooks';
import Subscription from '../Subscription';

const styles = {
  divider: css({
    height: 1,
    width: 'calc(100% + 32px)',
    backgroundColor: 'rgb(234, 234, 234)',
    marginLeft: -16,
    marginRight: -16,
    marginBottom: 16,
  }).toString(),
  emptyText: css({
    marginBottom: 16,
    textAlign: 'center',
  }).toString(),
  listTitle: css({
    fontWeight: '700',
  }).toString(),
};

/**
 * The Back In Stock Subscriptions List.
 * @returns {JSX}
 */
const List = () => {
  const {
    subscriptions,
    isInitial,
  } = useBackInStockSubscriptionsContext();

  const groupedSubscriptions = subscriptions.reduce((acc, subscription) => {
    const { status } = subscription;
    const groupingStatus = (status === 'inactive' || status === 'triggered') ? 'past' : status;
    acc[groupingStatus].push(subscription);
    return acc;
  }, {
    active: [],
    past: [],
  });

  // eslint-disable-next-line require-jsdoc
  const renderLabel = groupKey =>
    <div className={styles.listTitle}>
      {i18n.text(`back_in_stock.list_states.${groupKey}`)}
    </div>;

  return (
    <div>
      {Object.entries(groupedSubscriptions).map(([groupKey, filteredSubscriptions]) => (
        <Card className={styles.root} key={groupKey}>
          <Accordion
            className=""
            openWithChevron
            renderLabel={() => renderLabel(groupKey)}
            chevronPosition="left"
            startOpened
          >
            <div className={styles.divider} />
            {isInitial ? <LoadingIndicator /> : null}
            {!isInitial && filteredSubscriptions.length === 0 ? (
              <div className={styles.emptyText}>{i18n.text('back_in_stock.empty_list_reminder')}</div>
            ) : null}

            {!isInitial && filteredSubscriptions.map((subscription, index) => (
              <div key={subscription.subscriptionCode}>
                <Subscription
                  subscription={subscription}
                  productId={subscription.productCode}
                  key={subscription.subscriptionCode}
                  group={groupKey}
                />
                {(index === filteredSubscriptions.length - 1) ?
                  null :
                  <div className={styles.divider} />}
              </div>
            ))}
          </Accordion>
        </Card>
      ))}

    </div>
  );
};

export default List;
