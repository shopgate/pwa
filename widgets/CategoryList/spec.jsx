import React from 'react';
import { mount } from 'enzyme';
import List from 'Components/List';
import { Unwrapped as CategoryListWidget } from './index';

jest.mock('@shopgate/pwa-common/components/Link', () => {
  /**
   * Mocked LinkComponent
   * @return {JSX}
   */
  const Link = () => <div />;
  return Link;
});

describe('<CategoryListWidget />', () => {
  it('should not render the CategoryListWidget', () => {
    const props = {
      fetchCategory: () => {},
      items: null,
      settings: {
        categoryNumber: '',
        headline: 'Yay Categories',
        showImages: false,
      },
    };

    const wrapper = mount(<CategoryListWidget {...props} />);

    expect(wrapper.find(List.Item).length).toBe(0);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the CategoryListWidget', () => {
    const props = {
      fetchCategory: () => {},
      items: [
        {
          id: '1',
          name: 'Headline',
          imageUrl: '/some/url',
        },
        {
          id: '2',
          name: 'Headline',
          imageUrl: '/some/url',
        },
      ],
      settings: {
        categoryNumber: '',
        headline: 'Yay Categories',
        showImages: false,
      },
    };

    const wrapper = mount(<CategoryListWidget {...props} />);

    expect(wrapper.find(List.Item).length).toBe(2);
    expect(wrapper).toMatchSnapshot();
  });
});
