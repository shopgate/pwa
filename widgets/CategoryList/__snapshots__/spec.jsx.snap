// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryListWidget /> should not render the CategoryListWidget 1`] = `
<CategoryListWidget
  fetchCategory={[Function]}
  getCategory={[Function]}
  items={null}
  settings={
    Object {
      "categoryNumber": "",
      "headline": "Yay Categories",
      "showImages": false,
    }
  }
/>
`;

exports[`<CategoryListWidget /> should render the CategoryListWidget 1`] = `
<CategoryListWidget
  fetchCategory={[Function]}
  getCategory={[Function]}
  items={
    Array [
      Object {
        "id": "1",
        "imageUrl": "/some/url",
        "name": "Headline",
      },
      Object {
        "id": "2",
        "imageUrl": "/some/url",
        "name": "Headline",
      },
    ]
  }
  settings={
    Object {
      "categoryNumber": "",
      "headline": "Yay Categories",
      "showImages": false,
    }
  }
>
  <div
    className="css-1tgouoh"
    data-test-id="categoryList"
  >
    <h3
      className="css-twewb4"
    >
      Yay Categories
    </h3>
    <ResponsiveContainer
      appAlways={true}
      appOnly={false}
      breakpoint="<=xs"
      webAlways={false}
      webOnly={false}
    >
      <CategoryListDefault
        categories={
          Array [
            Object {
              "id": "1",
              "imageUrl": "/some/url",
              "name": "Headline",
            },
            Object {
              "id": "2",
              "imageUrl": "/some/url",
              "name": "Headline",
            },
          ]
        }
        settings={
          Object {
            "categoryNumber": "",
            "headline": "Yay Categories",
            "showImages": false,
          }
        }
      >
        <SheetList
          className={
            Object {
              "data-css-tgxhyl": "",
            }
          }
          hasImages={false}
          testId={null}
        >
          <List
            className={
              Object {
                "data-css-tgxhyl": "",
              }
            }
            role="listbox"
          >
            <ul
              className={
                Object {
                  "data-css-tgxhyl": "",
                }
              }
              role="listbox"
            >
              <ListItem
                className="css-id5jvr css-171mywc"
                isSelected={false}
                key="child-0/.$1"
                role={null}
                tabIndex={0}
              >
                <li
                  className="css-1p5q8vq css-id5jvr css-171mywc css-16gj3o1"
                  data-test-id="listItem"
                  role={null}
                >
                  <div
                    className="css-1ckzvk7"
                    data-test-id={null}
                  >
                    <ForwardRef(WithForwardedRef(Item))
                      image={null}
                      key="1"
                      link="/category/31"
                      linkComponent={[Function]}
                      rightComponent={
                        <Image
                          alt={null}
                          animating={true}
                          backgroundColor="#f2f2f2"
                          className="css-j7qxnf"
                          forcePlaceholder={false}
                          highestResolutionLoaded={[Function]}
                          onError={null}
                          onLoad={null}
                          ratio={null}
                          resolutions={
                            Array [
                              Object {
                                "blur": 2,
                                "height": 50,
                                "width": 50,
                              },
                              Object {
                                "height": 440,
                                "width": 440,
                              },
                            ]
                          }
                          src="/some/url"
                          srcmap={null}
                          transition={null}
                        />
                      }
                      testId="Headline"
                      title="Headline"
                    >
                      <Item
                        className={null}
                        description={null}
                        forwardedRef={null}
                        image={null}
                        isDisabled={false}
                        isSelected={false}
                        link="/category/31"
                        linkComponent={[Function]}
                        linkState={null}
                        onClick={null}
                        rightComponent={
                          <Image
                            alt={null}
                            animating={true}
                            backgroundColor="#f2f2f2"
                            className="css-j7qxnf"
                            forcePlaceholder={false}
                            highestResolutionLoaded={[Function]}
                            onError={null}
                            onLoad={null}
                            ratio={null}
                            resolutions={
                              Array [
                                Object {
                                  "blur": 2,
                                  "height": 50,
                                  "width": 50,
                                },
                                Object {
                                  "height": 440,
                                  "width": 440,
                                },
                              ]
                            }
                            src="/some/url"
                            srcmap={null}
                            transition={null}
                          />
                        }
                        testId="Headline"
                        title="Headline"
                      >
                        <ForwardRef(WithForwardedRef(Glow))
                          className={null}
                          styles={
                            Object {
                              "hover": Object {
                                "boxShadow": "-20px 0 0 red, 20px 0 0 #f7f7f7",
                              },
                            }
                          }
                        >
                          <Glow
                            className={null}
                            color="#f7f7f7"
                            forwardedRef={null}
                            styles={
                              Object {
                                "hover": Object {
                                  "boxShadow": "-20px 0 0 red, 20px 0 0 #f7f7f7",
                                },
                              }
                            }
                          >
                            <div
                              className="css-13azwyo"
                              onClick={[Function]}
                            >
                              <div
                                className="css-1bmd5dz"
                                style={Object {}}
                              />
                              <TextLink
                                className={null}
                                href="/category/31"
                                onClick={null}
                                state={null}
                                tabIndex={0}
                              >
                                <Link
                                  className="css-u12qp7"
                                  href="/category/31"
                                  onClick={null}
                                  state={null}
                                  tabIndex={0}
                                  tag="a"
                                >
                                  <div />
                                </Link>
                              </TextLink>
                            </div>
                          </Glow>
                        </ForwardRef(WithForwardedRef(Glow))>
                      </Item>
                    </ForwardRef(WithForwardedRef(Item))>
                  </div>
                </li>
              </ListItem>
              <ListItem
                className="css-id5jvr css-171mywc"
                isSelected={false}
                key="child-1/.$2"
                role={null}
                tabIndex={0}
              >
                <li
                  className="css-1p5q8vq css-id5jvr css-171mywc css-16gj3o1"
                  data-test-id="listItem"
                  role={null}
                >
                  <div
                    className="css-1ckzvk7"
                    data-test-id={null}
                  >
                    <ForwardRef(WithForwardedRef(Item))
                      image={null}
                      key="2"
                      link="/category/32"
                      linkComponent={[Function]}
                      rightComponent={
                        <Image
                          alt={null}
                          animating={true}
                          backgroundColor="#f2f2f2"
                          className="css-j7qxnf"
                          forcePlaceholder={false}
                          highestResolutionLoaded={[Function]}
                          onError={null}
                          onLoad={null}
                          ratio={null}
                          resolutions={
                            Array [
                              Object {
                                "blur": 2,
                                "height": 50,
                                "width": 50,
                              },
                              Object {
                                "height": 440,
                                "width": 440,
                              },
                            ]
                          }
                          src="/some/url"
                          srcmap={null}
                          transition={null}
                        />
                      }
                      testId="Headline"
                      title="Headline"
                    >
                      <Item
                        className={null}
                        description={null}
                        forwardedRef={null}
                        image={null}
                        isDisabled={false}
                        isSelected={false}
                        link="/category/32"
                        linkComponent={[Function]}
                        linkState={null}
                        onClick={null}
                        rightComponent={
                          <Image
                            alt={null}
                            animating={true}
                            backgroundColor="#f2f2f2"
                            className="css-j7qxnf"
                            forcePlaceholder={false}
                            highestResolutionLoaded={[Function]}
                            onError={null}
                            onLoad={null}
                            ratio={null}
                            resolutions={
                              Array [
                                Object {
                                  "blur": 2,
                                  "height": 50,
                                  "width": 50,
                                },
                                Object {
                                  "height": 440,
                                  "width": 440,
                                },
                              ]
                            }
                            src="/some/url"
                            srcmap={null}
                            transition={null}
                          />
                        }
                        testId="Headline"
                        title="Headline"
                      >
                        <ForwardRef(WithForwardedRef(Glow))
                          className={null}
                          styles={
                            Object {
                              "hover": Object {
                                "boxShadow": "-20px 0 0 red, 20px 0 0 #f7f7f7",
                              },
                            }
                          }
                        >
                          <Glow
                            className={null}
                            color="#f7f7f7"
                            forwardedRef={null}
                            styles={
                              Object {
                                "hover": Object {
                                  "boxShadow": "-20px 0 0 red, 20px 0 0 #f7f7f7",
                                },
                              }
                            }
                          >
                            <div
                              className="css-13azwyo"
                              onClick={[Function]}
                            >
                              <div
                                className="css-1bmd5dz"
                                style={Object {}}
                              />
                              <TextLink
                                className={null}
                                href="/category/32"
                                onClick={null}
                                state={null}
                                tabIndex={0}
                              >
                                <Link
                                  className="css-u12qp7"
                                  href="/category/32"
                                  onClick={null}
                                  state={null}
                                  tabIndex={0}
                                  tag="a"
                                >
                                  <div />
                                </Link>
                              </TextLink>
                            </div>
                          </Glow>
                        </ForwardRef(WithForwardedRef(Glow))>
                      </Item>
                    </ForwardRef(WithForwardedRef(Item))>
                  </div>
                </li>
              </ListItem>
            </ul>
          </List>
        </SheetList>
      </CategoryListDefault>
    </ResponsiveContainer>
    <ResponsiveContainer
      appAlways={false}
      appOnly={false}
      breakpoint=">xs"
      webAlways={false}
      webOnly={true}
    />
  </div>
</CategoryListWidget>
`;
